pkg/autodiscovery/listeners/snmp.go

```diff
@@ -228,7 +228,8 @@ func (l *SNMPListener) checkDevices() {
 	discoveryTicker := time.NewTicker(time.Duration(l.config.DiscoveryInterval) * time.Second)
 
 	for {
-		for _, subnet := range subnets {
+		for i := range subnets {
```

Comment:
`go` newbie question: being `subnets` [an array](https://github.com/DataDog/datadog-agent/pull/6943/files#diff-e58a1523f3120ce9446a8908240ffca2d36a095cdf748d67721dd71495621f3bR173) how is this different than previous code? Can't see how iterating with only indexes would be different than iterating with `key|value`

Reply:
@pducolin In the old code, the subnet have a same pointer address for each loop.

Example:  https://play.golang.org/p/bblg_R4AHqO

Reply:
Made a small improvement here: https://github.com/DataDog/datadog-agent/pull/6943/commits/baf55fab434124ed54818595fc90dbdb6d664c88

See also: https://play.golang.org/p/WNgEyUntm34

Reply:
Ah! This is a nice one! So range iteration gives a new pointer to the same value. Really interesting - and hard to catch.